bftengine/tests/bcstatetransfer/bcstatetransfer_tests.cpp

```diff
@@ -531,6 +531,26 @@ class BcStTest : public ::testing::Test {
   void onTimerImp() { stateTransfer_->onTimerImp(); }
   SourceSelector& getSourceSelector() { return stateTransfer_->sourceSelector_; }
 
+  bool compareSourceSelectorMetricKeyVal(const std::string& key, uint64_t val) {
+    if (key == "total_replacements_") {
+      return (getSourceSelector().metrics_.total_replacements_.Get().Get() == val);
+    } else if (key == "replacement_due_to_periodic_change_") {
+      return (getSourceSelector().metrics_.replacement_due_to_periodic_change_.Get().Get() == val);
+    } else if (key == "replacement_due_to_source_same_as_primary_") {
+      return (getSourceSelector().metrics_.replacement_due_to_source_same_as_primary_.Get().Get() == val);
+    } else if (key == "replacement_due_to_retransmission_timeout_") {
+      return (getSourceSelector().metrics_.replacement_due_to_retransmission_timeout_.Get().Get() == val);
+    } else {
+      EXPECT_EQ("Unexpected key", "");
+    }
+    return false;
+  }
+
+  bool checkDefaultSourceSelectorMetricValues() {
+    return ((getSourceSelector().metrics_.replacement_due_to_bad_data_.Get().Get() == 0) &&
+            (getSourceSelector().metrics_.replacement_due_to_retransmission_timeout_.Get().Get() == 0));
+  }
+
```

Comment:
If you look at line 561, there is a dedicated place for defintions: "BcStTest - definition".
Can you please declare functions here, and move definition later down?